src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java

```diff
@@ -189,6 +187,80 @@ public void shouldUpdateCcdTwiceGivenFileWithFurtherEvidence() throws Exception
             .update(any(CaseData.class), anyLong(), eq("appealReceived"), eq(idamTokens));
     }
 
+    @Test
+    public void shouldAddExistingHearingDetailsToTheCaseIfItsMissingInComingGaps2Xml() throws Exception {
+
+        ArgumentCaptor<CaseData> caseDataArgumentCaptor = ArgumentCaptor.forClass(CaseData.class);
+
+        CaseData caseData = buildCaseData(RESPONSE_RECEIVED);
+        caseData.setHearings(buildHearings());
+
+        CaseDetails existingCaseDetails = getCaseDetails(CASE_DETAILS_WITH_HEARINGS_JSON);
+
+        ccdCasesSender.sendUpdateCcdCases(caseData, existingCaseDetails, idamTokens);
+
+        verify(updateCcdService).update(caseDataArgumentCaptor.capture(),
+            eq(existingCaseDetails.getId()), eq(caseData.getLatestEventType()), eq(idamTokens));
+
+        assertThat(caseDataArgumentCaptor.getValue().getHearings().size(), equalTo(2));
+        assertThat(caseDataArgumentCaptor.getValue().getHearings().get(0).getValue().getHearingDateTime(),
+            equalTo("2018-04-0510:00:00"));
+        assertThat(caseDataArgumentCaptor.getValue().getHearings().get(1).getValue().getHearingDateTime(),
+            equalTo("2017-05-2410:00"));
+
+
+    }
+
+    @Test
+    public void shouldAddNewHearingDetailsFromGap2XmlToTheCcd() throws Exception {
+
+        ArgumentCaptor<CaseData> caseDataArgumentCaptor = ArgumentCaptor.forClass(CaseData.class);
+
+        CaseData caseData = buildCaseData(RESPONSE_RECEIVED);
+        caseData.setHearings(buildHearings());
+
+        CaseDetails existingCaseDetails = getCaseDetails(CASE_DETAILS_WITH_NO_HEARINGS_JSON);
+
+        ccdCasesSender.sendUpdateCcdCases(caseData, existingCaseDetails, idamTokens);
+
+        verify(updateCcdService).update(caseDataArgumentCaptor.capture(),
+            eq(existingCaseDetails.getId()), eq(caseData.getLatestEventType()), eq(idamTokens));
+
+        assertThat(caseDataArgumentCaptor.getValue().getHearings().size(), equalTo(1));
+        assertThat(caseDataArgumentCaptor.getValue().getHearings().get(0).getValue().getHearingDateTime(),
+            equalTo("2018-04-0510:00:00"));
+
+    }
+
+    @Test
+    public void shouldAddExistingNewHearingDetailsFromCcdToCase() throws Exception {
```

Comment:
Any chance of adding a `when` to the test name so we know straight away what exactly id being tested? (Same for test above). e.g. shouldAddExistingNewHearingDetailsFromCcdToCaseWhenHearingDetailsAreEmpty are similar.

Reply:
guys, it would be good to follow some naming convention for our test methods. For example, the one that Jack is suggestiong or this other one -> givenCondition_shouldExpected

Reply:
Yep, @sathishbabuj I think the `givenCondition_shouldExpected` is the convention used for case-loader

Reply:
Its just a personnel preference. As long as the test name is defined fine it should be ok. @jackmaloney its not a convention in case loader. Please check CaseLoaderServiceTest.

Reply:
`CaseLoaderServiceTest` was created by Aaron so was probably missed. Anyway, my point was I found trying to work out the difference between those 2 tests confusing without studying the code. So for clarity would you be able to improve the test name please?

Reply:
@sathishbabuj the part missing in the test name is the condition, when that should happen, given what, under what circumstances. Adding the give/when would make clearer.

Reply:
added When condition to the test name